PayloadPkg/FirmwareUpdate/CsmeFwUpdate.c

```diff
@@ -130,9 +130,9 @@ CsmePciReadBuffer (
     return EFI_INVALID_PARAMETER;
   }
 
-  Bus = (UINT8)((StartAddress >> 24) & 0xFF);
-  Device = (UINT8)((StartAddress >> 16) & 0xFF);
-  Function = (UINT8)((StartAddress >> 8) & 0xFF);
+  Bus = (UINT8)((RShiftU64(StartAddress, 24)) & 0xFF);

```

Comment:
since upper 32 bit is not used at all for Bus/Device/Function calculation.
I would recommend to use a new UINT32 to hold the PciAddress.  Then it is safe to do direct shift without using RShiftU64.
PciAddress = (UINT32)StartAddress;
Bus = (UINT8)((PciAddress >> 24) & 0xFF);